modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/ComponentMessageProcessor.java

```diff
@@ -176,48 +181,66 @@ public CoreEvent process(CoreEvent event) throws MuleException {
+              OperationExecutionFunction operationExecutionFunction;
+              final Scheduler currentScheduler =
+                  ctx.getOrEmpty(PROCESSOR_SCHEDULER_CONTEXT_KEY).map(s -> (Scheduler) s).orElse(IMMEDIATE_SCHEDULER);
+
+              final PrecalculatedExecutionContextAdapter<T> precalculatedEvent = getPrecalculatedContext(eventWithContext);
+              if (getLocation() != null && isInterceptedComponent(getLocation(), (InternalEvent) eventWithContext)
+                  && precalculatedEvent != null) {
+                ExecutionContextAdapter<T> operationContext = getPrecalculatedContext(eventWithContext);
+
+                operationExecutionFunction = (parameters, operationEvent) -> {
+                  operationContext.setCurrentScheduler(currentScheduler);
+                  return doProcessWithErrorMapping(operationEvent, operationContext);
+                };
+              } else {
+                operationExecutionFunction = (parameters, operationEvent) -> {
+                  ExecutionContextAdapter<T> operationContext;
+                  try {
+                    operationContext = createExecutionContext(configuration, parameters, operationEvent, currentScheduler);
+                  } catch (MuleException e) {
+                    return error(e);
+                  }
+                  return doProcessWithErrorMapping(operationEvent, operationContext);
+                };
+              }
 
-          if (getLocation() != null) {
-            ((DefaultFlowCallStack) event.getFlowCallStack()).setCurrentProcessorPath(resolvedProcessorRepresentation);
-            return policyManager
-                .createOperationPolicy(this, event, () -> resolutionResult)
-                .process(event, operationExecutionFunction, () -> resolutionResult, getLocation());
-          } else {
-            // If this operation has no component location then it is internal. Don't apply policies on internal operations.
-            return operationExecutionFunction.execute(resolutionResult, event);
-          }
-        }));
+              if (getLocation() != null) {
+                ((DefaultFlowCallStack) eventWithContext.getFlowCallStack())
+                    .setCurrentProcessorPath(resolvedProcessorRepresentation);
+                return Mono.from(policyManager
+                    .createOperationPolicy(this, eventWithContext, () -> resolutionResult)
+                    .process(eventWithContext, operationExecutionFunction, () -> resolutionResult, getLocation()));
+              } else {
+                // If this operation has no component location then it is internal. Don't apply policies on internal operations.
+                return Mono.from(operationExecutionFunction.execute(resolutionResult, eventWithContext));
+              }
+            })));
+  }
+
+  private CoreEvent addContextToEvent(CoreEvent event, Context ctx) {
+    final CoreEvent eventWithContext;
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this unused "eventWithContext" local variable. ([squid:S1481](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS1481))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7661&issues=AWjOiDKKotIXnmsUmtMc&open=AWjOiDKKotIXnmsUmtMc)

[ISSUE_KEY]: AWjOiDKKotIXnmsUmtMc